### PR TITLE
skus sdk - attempt to fetch missing credentials prior to credential summary

### PR DESCRIPTION
**Pull request by husobee**
Resolves 

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/skus/browser/rs/lib/src/sdk/credentials/mod.rs [hunk 1]
```diff
@@ -37,6 +37,15 @@ where
             return Err(InternalError::OrderLocationMismatch.into());
         }
 
+        // if we have no credentials at all, we should attempt to
+        // fetch_order_credentials to populate prior to trying to 
+        // presenting the order cred summary to the caller, otherwise we 
+        // will get an InternalError::NotFound
+        if !self.client.has_credentials(&order.id).await? {
+            self.fetch_order_credentials(order_id).await?;
```
Comment (bsclifton) [inline components/skus/browser/rs/lib/src/sdk/credentials/mod.rs:45]: Can this still fail the same way? Just curious if we should actually catch/handle the error (along with this change)

Comment (husobee) [reply components/skus/browser/rs/lib/src/sdk/credentials/mod.rs:45]: Yes, potentially it can still fail the same way.  The root of the problem is the below:
https://github.com/brave/brave-core/blob/bh-skus-sdk-fetch-creds-if-absent-summary/components/skus/browser/rs/cxx/src/lib.rs#L476-L477

Returning an empty string "" as a summary is breaking the below:
https://github.com/brave/brave-core/blob/bh-skus-sdk-fetch-creds-if-absent-summary/components/skus/renderer/skus_js_handler.cc#L291-L300

I would maybe suggest we change this to bubble up the underlying skus sdk error to the JS value like we do in the WASM bindings, or pass back an empty object.  In the accounts reference site implementation it is performing a check on if there are credentials for a given domain first in order to check if it needs to perform a provision.

Basically without performing an initial provision call on the skus sdk, any JS code calling credential_summary will fail, this change is attempting to intercept this condition and make a best effort to fetch order credentials first.

We could fix the error propogation, or change the `result_v` check in the renderer to possibly send back an empty object which will allow the JS code to handle the issue appropriately.

**Review by bsclifton — APPROVED** [review]

File: components/skus/browser/rs/lib/src/sdk/credentials/mod.rs (hunk 1 above)
Comment (bsclifton) [reply components/skus/browser/rs/lib/src/sdk/credentials/mod.rs:45]: Pushed a fix for empty string

Comment (bsclifton) [reply components/skus/browser/rs/lib/src/sdk/credentials/mod.rs:45]: hmm - either my change or this change is breaking unit tests. Taking a look at this

**Comment by husobee** [comment]
Closing for now, leaving the caller of the native sdk client to handle the exception as exists currently.
